### ISSUE TITLE
rustdoc sidebar is not accessible from keyboard on small screens

### ISSUE THREAD

**Issue by notriddle** [rust-lang/rust#87172]
On a PC (or on an iPad with "Full Keyboard Access" turned on), shrink the browser so that the sidebar goes into the overflow menu, then try to get at it with a keyboard.

There's no focus ring. You can actually get keyboard focus on it, in iPadOS, but you can't see it, and I can't seem to get keyboard access to it at all on desktop Firefox. In either case, it's not good enough. The hamburger button is never highlighted, because it is not keyboard focusable (it's a DIV, and not a real button). #87167 won't make this better, since if you are using the keyboard, but are not using a screenreader, the menu items will become focused but you won't be able to see them.

To combine this with #87167, JavaScript can be used to make sidebar visibility get automatically switched based on focus (if a menu item is focused, make it visible, and if the focus goes onto something outside the menu, hide it).
